validator/keymanager/remote-web3signer/internal/client.go

```diff
@@ -151,6 +149,12 @@ func (client *ApiClient) doRequest(ctx context.Context, httpMethod, fullPath str
 	} else {
 		signRequestDurationSeconds.WithLabelValues(req.Method, strconv.Itoa(resp.StatusCode)).Observe(duration.Seconds())
 	}
+	if resp.StatusCode != http.StatusOK {
+		requestDump, err = httputil.DumpRequest(req, true)
```

Comment:
I read the godoc for this method and it mentions
> It should only be used by servers to debug client requests.

Is there a more appropriate way to print the request? Why not just print the body in the debug error?

Reply:
i was trying to follow this https://medium.com/doing-things-right/pretty-printing-http-requests-in-golang-a918d5aaa000. I believe this was because the request could have pointers to objects and we need to print it in a pretty understandable way recursively to print out the whole request object. I thought it was appropriate because we are sort of debugging the client's requests since it's a webhook, it's just we are running it on their behalf.

Reply:
improved it with latest commit